### ISSUE TITLE
Argument for parameter does not exist and is missing at the same time.

### ISSUE THREAD

**Issue by PietervdStar** [doxygen/doxygen#10123]
[edit]I was completely misled when I wrote this issue, actual issue is Doxygen seeing calls to this function as a declaration. For more info see [my first comment](https://github.com/doxygen/doxygen/issues/10123#issuecomment-1586277900).
[/edit]

I have a JavaScript file without any class and happened upon a function name that somehow breaks the parameter list. Either I get a warning about not all parameters have been documented, or I get a warning that the argument for `@parameter` does not exist.
This is only if I also have an `@file` at the start, and have only found one function name (dynamically_load_js) that triggers this issue.

Error message I get is either:
`error: argument 'myparameter' of command @param is not found in the argument list of dynamically_load_js(myparameter) (warning treated as error, aborting now)`
or
` error: parameters of member dynamically_load_js are not (all) documented (warning treated as error, aborting now)`

I expect Doxygen to recognise the argument from the argument list and see the `@param` documentation for it. As with all other functions in the minimal example below, or when the `@file` is absent. 

I can probably find a different name for the function as a workaround.

Using version 1.9.1 on Ubuntu 22.04.2 LTS

```JavaScript
/** @file
 */

/** This is a dummy function for the minimal example that works.
 *  @param myparameter This parameter definitely is in the parameter list.
 *  @returns Nothing.
 */
function dynamically_load_js_(myparameter) {
	console.log(myparameter);
}

/** This is another dummy function for the minimal example that works.
 *  @param myparameter This parameter definitely is in the parameter list.
 *  @returns Nothing.
 */
function ynamically_load_js_(myparameter) {
	console.log(myparameter);
}

/**dummy class*/
class dummy{
	/** This is another dummy function for the minimal example that works.
	*  @param myparameter This parameter definitely is in the parameter list.
	*  @returns Nothing.
	*/
	function dynamically_load_js(myparameter) {
		console.log(myparameter);
	}
}

/** This is a dummy function for the minimal example that doesn't work.
 *  @param myparameter This parameter defenitely is in the parameter list.
 *  @returns Nothing.
 */
function dynamically_load_js(myparameter) {
	console.log(myparameter);
}
```

[Doxyfile.txt](https://github.com/doxygen/doxygen/files/11715409/Doxyfile.txt)

**Comment by PietervdStar** [comment]
hmm, got some additional information that makes the problem make more sense. 
The problem actually lies in me calling that function from other files. It is basically the first non-comment line in these files and there I use something like `dynamically_load_js("somestring")`. This probably confuses Doxygen and it sees that as the declaration. So the workaround mentioned above doesn't solve the issue, and the example is not an example of the actual problem. 

The problem is Doxygen seeing the function calls as declaration(s?). As a workaround I wrapped those calls up in a function with a unique name for each file.

**Comment by albert-github** [comment]
You wrote "he problem actually lies in me calling that function from other files.", but the example looks like the have just 1 file, so to be able to reproduce the problem:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

Furthermore:
- doxygen 1.9.1 is from January 8, 2023, the current doxygen version is 1.9.7 (and there are probably some improvements regarding warnings).
- javascript is not ab actively supported language by doxygen, which might result in some "inconveniences"

**Comment by PietervdStar** [comment]
Yeah, I know about the limited support, but haven't found a better alternative yet :)
I can try to upgrade but that would mean ditching the version from my package manager, I will look in to that, hopefully later this week. 
The small attached directory will have to wait as there it doesn't show, so I'll have to fiddle with it to see if I can get it to reproduce the issue from there. A tar will be coming your way in the upcoming days.

**Comment by PietervdStar** [comment]
Github doesn't seem to like tar files, so here is a zip
[issue_10123.zip](https://github.com/doxygen/doxygen/files/11726849/issue_10123.zip)
 instead.

**Comment by albert-github** [comment]
Also with the doxygen version 1.9.7 I see the warning:
```
js/base.js:5: error: argument 'path' of command @param is not found in the argument list of dynamically_load_js(path) (warning treated as error, aborting now)
```

I never noticed that github has problems with uploading `tar` files but I always upload compressed tar files created like `tar czf example.tar,gz js` (instead of `tar cf example.tar js`)

**Comment by albert-github** [comment]
In the `js.base.js` we have:
```
function dynamically_load_js(path) {
        console.log(path);
}
```
which is a regular function definition and also handled as such.
in `js/types.js` we have:
```
dynamically_load_js("one_two");
```
and this is an invocation of `dynamically_load_js` but it is seen as a function prototype / the real definition by doxygen.
When I rename the `js/types.js` to `js/a_types.js`  there is no warning anymore (doxygen probably sees the function in `js/base.js` as the proper function).
(The rename was just for the analysis).

@doxygen Any idea?

**Comment by PietervdStar** [comment]
I just got an idea, can't test this just now but should be able to this weekend. What if I specify base as an `@headerfile`? I'll try and report back later.

**Comment by PietervdStar** [comment]
as usual plans change, became today instead of last week, but nope, didn't make a difference.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10166

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by PietervdStar** [comment]
Thanks.

Quick question, is doxygen at googlemail still the contact address for the project?

**Comment by albert-github** [comment]
That is the official email, but it is better to pose questions etc. at stack overflow and here.
The email address is solely used by @doxygen.
Use the email address only for direct questions / remarks etc. that are not suited for public view.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
